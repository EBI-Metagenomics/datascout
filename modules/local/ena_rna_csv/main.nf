#!/usr/bin/env nextflow
process ENA_RNA_CSV {

    container 'community.wave.seqera.io/library/python_pip_biopython_requests:725bda83fb97ec48'
    debug true
    publishDir "${params.output}", mode: "copy"

    errorStrategy 'retry'
    maxRetries 2
    
    input:
      tuple val(meta), file(tax_ranks)
      tuple val(meta), val(rank)

    output:
    tuple val(meta), path("${meta.id}_ENA_filtered_rna.csv"), emit: rna_csv

    script:
    prefix = meta.id
    """
    rna_seq.py --tax_file ${tax_ranks} --output_file "${prefix}_ENA_filtered_rna.csv" --rank ${rank}
    """
}

// Get transcriptome metadata from ena, filtered and reordered by sample name