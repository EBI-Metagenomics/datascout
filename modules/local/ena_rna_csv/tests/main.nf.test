nextflow_process {

    name "Test Process ENA_RNA_CSV"
    script "modules/local/ena_rna_csv/main.nf"
    process "ENA_RNA_CSV"

    test("Should run without failures") {

        when {

            process {
                """
                input[0] = tuple([id:"test_sample1", genome_id:"GCA_003719455.1", ena_tax:"default"], "${projectDir}/tests/reference_databases/test_sample1_tax_ranks.tsv")
                input[1] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.rna_csv).match()
        }

    }

}
