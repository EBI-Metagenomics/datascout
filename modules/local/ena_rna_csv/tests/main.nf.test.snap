{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_sample1",
                        "genome_id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "GCA_003719455.1_ENA_filtered_rna.csv:md5,15cad058a4845db7b543f89e6bb52ebb"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-17T11:19:24.703001"
    }
}